misc/python/materialize/zippy/source_actions.py

```diff
@@ -79,3 +101,24 @@ def run(self, c: Composition) -> None:
 
     def provides(self) -> list[Capability]:
         return [self.source]
+
+
+class AlterSource(Action):
+    def __init__(self, capabilities: Capabilities, source: SourceExists) -> None:
+        self.source = source
+        super().__init__(capabilities)
+
+    def run(self, c: Composition) -> None:
+        c.testdrive(
+            dedent(
+                f"""
+                > ALTER CONNECTION kafka_conn_alter_connection_2 SET BROKER '${{testdrive.kafka-addr}}'
```

Comment:
I fear that this doesn't test much as it is essentially a nop

Reply:
What exactly do you mean? That this statement as such is a NOP or that the test is a NOP?
If it is about the statement, then I will need to come up with something else. If it is about the test, I want to highlight that this is only a piece that is inserted into a sequence of actions.

Reply:
This `ALTER` action sets the connection to the same value it already has--that's what I mean by NOP. It doesn't change any behavior--e.g. if this statement only parsed and returned an OK, this test would not uncover that.

Reply:
My understanding is that it will change the connection to `USING SSH TUNNEL thancred` or drop that from the connection definition. That is, it is supposed to change to the respective opposite (note the `not` in `if not self.source.uses_ssh_tunnel`).

Or do I miss something?

Reply:
Oh, so sorry--I missed [this line](https://github.com/MaterializeInc/materialize/pull/23107/files#diff-4b4e49e5cd414d9531c8acbcdaab15ff12796c41a3684fabbc2e9ededf89a11eR121) when reading and didn't see it getting toggled back and forth!

Reply:
No, problem!

Reply:
I should maybe add a comment to make that more obvious.